Source/WTF/wtf/SegmentedVector.h

```diff
@@ -265,15 +264,13 @@ namespace WTF {
 
         void allocateSegment()
         {
-            m_segments.append(static_cast<Segment*>(Malloc::malloc(sizeof(T) * SegmentSize)));
+            m_segments.append(MallocPtr<Segment, Malloc>::malloc(sizeof(T) * SegmentSize));
         }
 
         size_t m_size { 0 };
-        Vector<Segment*, 0, CrashOnOverflow, 16, Malloc> m_segments;
+        Vector<MallocPtr<Segment>, 0, CrashOnOverflow, 16, Malloc> m_segments;
```

Comment:
Can we find a way to use something more like `std::unique_ptr<Segment[]>` instead of `MallocPtr`? I ask because `MallocPtr.h` says “do not use this in new code”.